load-test-framework/dotnet_src/Google.PubSub.Flic/Program.cs

```diff
@@ -0,0 +1,225 @@
+            {
+                lock (_lock)
+                {
+                    var ret = new CheckResponse
+                    {
+                        BucketValues = { _histogram.Reverse().SkipWhile(x => x == 0).Reverse() },
+                        Failed = _errorCount,
+                        IsFinished = _endTime != null,
+                        RunningDuration = Duration.FromTimeSpan(_startTime is DateTime dt ? (_endTime ?? DateTime.UtcNow) - dt : TimeSpan.Zero),
+                        ReceivedMessages = { _msgIds.Select(x => new MessageIdentifier { PublisherClientId = _clientId, SequenceNumber = x }) },
+                    };
+                    _errorCount = 0;
+                    Array.Clear(_histogram, 0, _histogram.Length);
+                    _msgIds.Clear();
+                    return ret;
+                }
+            }
+        }
+
+        static async Task<int> Main(string[] args)
+        {
+            var parsed = Parser.Default.ParseArguments<Options>(args);
+            switch (parsed)
+            {
+                case Parsed<Options> success:
+                    var options = success.Value;
+                    var startTcs = new TaskCompletionSource<StartRequest>();
+                    var stats = new Stats();
+                    Console.WriteLine($"Starting dotnet server on port:{options.Port}");
+                    var server = new Server
+                    {
+                        Services = { LoadtestWorker.BindService(new LoadtestServer(startTcs, stats.BuildCheckResponse)) },
+                        Ports = { new ServerPort("0.0.0.0", options.Port, ServerCredentials.Insecure) },
+                    };
+                    server.Start();
+                    var startRequest = await startTcs.Task;
+                    Console.WriteLine($"  Start: Test duration: {(int)startRequest.TestDuration.ToTimeSpan().TotalSeconds}s; include ids:{startRequest.IncludeIds}; CPU scaling:{startRequest.CpuScaling}");
+                    var now = DateTime.UtcNow;
+                    stats.Init(now, startRequest.IncludeIds);
+                    var waitDuration = startRequest.StartTime.ToDateTime() - now;
+                    if (waitDuration > TimeSpan.Zero)
+                    {
+                        await Task.Delay(waitDuration);
+                    }
+                    switch (startRequest.ClientOptionsCase)
+                    {
+                        case StartRequest.ClientOptionsOneofCase.PublisherOptions:
+                            await Publish(startRequest, stats);
+                            break;
+                        case StartRequest.ClientOptionsOneofCase.SubscriberOptions:
+                        default: // no option is sent for subscribe
+                            await Subscribe(startRequest, stats);
+                            break;
+                    }
+                    await Task.Delay(TimeSpan.FromHours(6));
```

Comment:
why are you doing this?

Reply:
I should clarify: why are you sleeping for 6 hours here? Isnt there a method on the grpc server to sleep the main thread so the program keeps running while the server is serving?